llarp/service/outbound_context.cpp

```diff
@@ -407,6 +405,18 @@ namespace llarp::service
 
     std::vector<Introduction> intros = current_intro.intros;
 
+    // don't consider intros for which we don't have the RC for the pivot
+    auto itr = intros.begin();
+    while (itr != intros.end())
+    {
+      if (not ep.router()->node_db()->has_router(itr->router))
```

Comment:
Minor:
Maybe we can bring the call to `NodeDB::has_router` into a method in `Router` that describes what you're checking, like `Router::has_peer_router`? (or something less dumb lol)

Reply:
The method name here is dumb, agreed.  As for having a method that simply passes through the request to a sub-object: it's something I've done quite a bit in the past and find it's almost never worth the abstraction if it's safe/sane for the caller to just use the sub-object directly.  Either way, either this should change to something like you suggest here, or `router_whitelist` (rename pending) above (and probably others) should go away and callers should do `router->node_db()->whitelist()` if they need it.  I prefer the latter; thoughts?

Reply:
I don't feel super strongly about it, except that if we're doing `ep.router().whatever()->whatever_method()->whatever2->yadda_yadda()` more than once in a file maybe we should method it?